### PR TITLE
Common worker abstraction

### PR DESCRIPTION
**Pull request by pavletto**
INSERT A SMALL DESCRIPTION OF WHAT YOU CHANGED HERE

#### **Verify that...**

- [x] `npm test` passes
- [x] `npm run api` passes
- [x] `npm run lint` passes
- [x] `npm run build` still works

### CODE REVIEW

File: src/og/control/visibleExtent/Segment.js [hunk 1]
```diff
@@ -403,8 +403,9 @@ class Segment {
      * @param {Float32Array} elevations - Elevation data.
      */
     elevationsExists(elevations) {
+        const segment = this;
         if (this.plainReady && this.terrainIsLoading) {
-            this.planet._terrainWorker.make(this, elevations);
+            this.planet._terrainWorker.make({segment, elevations});
```
Comment (Zemledelec) [inline src/og/control/visibleExtent/Segment.js]: dont like instancing.

File: src/og/entity/LabelHandler.js [hunk 2]
```diff
@@ -85,7 +85,8 @@ class LabelHandler extends BillboardHandler {
     }
 
     _addLabelToArrays(label) {
-        this._renderer.labelWorker.make(this, label);
+        const handler = this;
+        this._renderer.labelWorker.make({ handler, label });
```
Comment (Zemledelec) [inline src/og/entity/LabelHandler.js]: the same don't like instancing

File: src/og/entity/LabelWorker.js [hunk 3]
```diff
@@ -2,76 +2,59 @@
...
-        if (this._pendingQueue.length) {
-            var p = this._pendingQueue.pop();
-            this.make(p.handler, p.label);
-        }
+        this.check();
     }
 
-    make(handler, label) {
 
+    make({ handler, label }) {
```
Comment (Zemledelec) [inline src/og/entity/LabelWorker.js]: There is a IoC possible...thinking...

Comment (Zemledelec) [reply src/og/entity/LabelWorker.js]: ...maybe it could be refactored in a next iteration

File: src/og/segment/Segment.js [hunk 4]
```diff
@@ -386,8 +386,9 @@ class Segment {
      * @param {Float32Array} elevations - Elevation data.
      */
     elevationsExists(elevations) {
+        const segment = this;
         if (this.plainReady && this.terrainIsLoading) {
-            this.planet._terrainWorker.make(this, elevations);
+            this.planet._terrainWorker.make({ segment, elevations });
```
Comment (Zemledelec) [inline src/og/segment/Segment.js]: don't like instancing

**Review by Zemledelec — CHANGES_REQUESTED** [review]

File: src/og/segment/Segment.js (hunk 4 above)
Comment (pavletto) [reply src/og/segment/Segment.js]: fix that

**Comment by pavletto** [comment]
@Zemledelec check my fixes, please

**Review by Zemledelec — APPROVED** [review]

**Pull request merged**